plugins/ingest/src/main/java/org/elasticsearch/ingest/Pipeline.java

```diff
@@ -65,7 +65,7 @@ public Builder(String name) {
             this.name = name;
         }
 
-        public Builder(Map<String, Object> config) {
+        public Builder(Map<String, Object> config, Map<String, Processor.Builder> processorRegistry) {
```

Comment:
If we move to prototypes than that means that we a processor can only built with a map of maps. I though that also the idea was to have a real builder that has setters for its properties. So maybe we should then take the builder & factory direction?